### PR TITLE
feat(client): Fully validate `email` and `uid` in the relier.

### LINKED ISSUE #3650: fully validate email query param
In #3606 we started fully validating the email query parameter in the `/force_auth` flow. Before #3606, force_auth acted differently to other views when an email is invalid, it printed its own error message. Now that we have unified behaviors, we should push the validation into [models/relier.js](https://github.com/mozilla/fxa-content-server/blob/b889bdfd5e36941c4de88eddddb5e76add418daa/app/scripts/models/reliers/relier.js#L41) so that it's available to all views.

Note, there is no HTML injection attack with an invalid email. The invalid email will be displayed to the user in an input field, which may cause confusion.

### PR DESCRIPTION
**Pull request by shane-tomlinson**
fixes #3650

@philbooth - r?

### CODE REVIEW

**Comment by shane-tomlinson** [comment]
or @vladikoff or @vbudhram if you are still online, if not Phil will be online earlier than both of you tomorrow.

File: tests/functional/settings.js [hunk 1]
```diff
@@ -94,43 +107,6 @@ define([
         .then(FunctionalHelpers.noSuchElement(self, '#signout'));
     },
 
-    'visit settings page with an invalid sessionToken redirects to signin': function () {
```
Comment (shane-tomlinson) [inline tests/functional/settings.js]: These tests were duplicated in settings_common, so I ripped them out.

**Comment by shane-tomlinson** [comment]
Putting WIP on this, I want to add more functional tests.

**Comment by shane-tomlinson** [comment]
Pulling WIP back off of this, I wanted to check for "damaged link" tests, many already exist in tests/functional/complete_sign_up.js and tests/functional/reset_password.js

**Comment by philbooth** [comment]
LGTM 👍

**Pull request merged**